app/controllers/api/service_dialogs_controller.rb

```diff
@@ -15,8 +22,27 @@ def refresh_dialog_fields_resource(type, id = nil, data = nil)
 
     private
 
-    def set_additional_attributes
-      @additional_attributes = %w(content) if attribute_selection == "all"
+    def single_resource(resource, params, type = :service_dialogs, reftype = :service_dialogs)
+      json = resource_to_jbuilder(type, reftype, resource)
+      expand_dialog_content(json, resource) unless params.key?(:attributes)
+      json
+    end
+
+    def dialog_collection(params = {})
+      dialogs = Dialog.all
+      Jbuilder.new do |json|
+        json.ignore_nil!
+        json.set! 'name', 'service_dialogs'
+        json.set! 'count', dialogs.count
+        json.set! 'subcount', dialogs.count
+        json.resources dialogs.collect do |resource|
+          if params['expand'] == 'resources'
+            add_hash json, single_resource(resource, params).attributes!
+          else
+            json.href normalize_href(:service_dialogs, resource['id'])
+          end
+        end
+      end
```

Comment:
same idea as above,  Jbuilder knowledge should only be in renderer.rb,  let's build the hash here then use the methods in renderer to do the Jbuilder magic.